samples/bluetooth/hap_ha/src/csip_set_member.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022 Codecoup
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/sys/printk.h>
+
+#include <zephyr/bluetooth/audio/cap.h>
+#include <zephyr/bluetooth/audio/csis.h>
+
+#define CSIS_SIRK_DEBUG	{ 0xcd, 0xcc, 0x72, 0xdd, 0x86, 0x8c, 0xcd, 0xce, \
+			  0x22, 0xfd, 0xa1, 0x21, 0x09, 0x7d, 0x7d, 0x45 }
+
+static struct bt_csis *csis;
+
+static void csis_lock_changed_cb(struct bt_conn *conn, struct bt_csis *csis,
+				 bool locked)
+{
+	printk("Client %p %s the lock\n", conn, locked ? "locked" : "released");
+}
+
+static uint8_t sirk_read_req_cb(struct bt_conn *conn, struct bt_csis *csis)
+{
+	return BT_CSIS_READ_SIRK_REQ_RSP_ACCEPT;
+}
+
+static void rsi_changed_cb(const uint8_t *rsi)
+{
+	char rsi_str[13];
+
+	snprintk(rsi_str, 163, "%02x%02x%02x%02x%02x%02x",
+		 rsi[0], rsi[1], rsi[2], rsi[3], rsi[4], rsi[5]);
+
+	printk("PRSI: 0x%s\n", rsi_str);
+}
```

Comment:
This should trigger a advertising data update and restart if the sample is advertising the RSI itself, rather than using `bt_csis_advertise`.

Reply:
Agree, but this requires further refactor in CSIS as well. Now, when `bt_le_ext_adv_cb` has `rpa_expired` callback, it would be good to have a function like `bt_csis_generate_rsi`. That would solve the issue, since, even if application manages the RSI itself, it has to call `bt_csis_advertise` anyway, to get the RSI updates. But if it knows when the RPA expired by `rpa_expired` callback then it would call API function to get new RSI itself